src/e2e/limits.rs

```diff
@@ -0,0 +1,189 @@
+use crate::contracts::{get_liquidity_by_x, get_liquidity_by_y, FeeTier, PoolKey};
+use crate::math::sqrt_price::calculate_sqrt_price;
+use crate::math::token_amount::TokenAmount;
+use crate::math::{percentage::Percentage, sqrt_price::SqrtPrice, MAX_SQRT_PRICE, MIN_SQRT_PRICE};
+use crate::token::{TokenDeployer, TokenRef};
+use crate::{InvariantDeployer, InvariantRef};
+use decimal::{Decimal, Factories};
+use odra::prelude::string::String;
+use odra::test_env;
+use odra::types::{U128, U256};
+
+fn init_dex_and_tokens_max_mint_amount() -> (InvariantRef, TokenRef, TokenRef) {
+    let alice = test_env::get_account(0);
+    test_env::set_caller(alice);
+
+    let mint_amount = U256::max_value();
+
+    let token_x = TokenDeployer::init(String::from(""), String::from(""), 0, &mint_amount);
+    let token_y = TokenDeployer::init(String::from(""), String::from(""), 0, &mint_amount);
+    let invariant = InvariantDeployer::init(Percentage::from_scale(1, 2));
+
+    (invariant, token_x, token_y)
+}
+
+fn big_deposit_and_swap(x_to_y: bool) {
```

Comment:
Also, decide if you need this function. In this case, the location is correct, but if you decide to rename this function, you should add this macro above: `#[cfg(test)]`